File: app/src/main/java/edu/tjrac/swant/todo/view/WebWorkSpaceActivity.java
package edu.tjrac.swant.todo.view;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.yckj.baselib.common.base.BaseActivity;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import edu.tjrac.swant.todo.adapter.WebWorkGroupAdapter;
import edu.tjrac.swant.todo.bean.WebInfo;
import edu.tjrac.swant.unicorn.R;

public class WebWorkSpaceActivity extends BaseActivity {

    @BindView(R.id.recycler) RecyclerView mRecycler;
//    @BindView(R.id.fl_note) FrameLayout mFlNote;


    List<WebWorkGroupAdapter.WebWorkGroupInfo> groups = new ArrayList<>();
    WebWorkGroupAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_web_work_space);
        ButterKnife.bind(this);
        WebWorkGroupAdapter.WebWorkGroupInfo webs = getIntent().getParcelableExtra("groups");
        if (webs != null) {
            groups.add(webs);
        }
        mRecycler.setLayoutManager(new GridLayoutManager(WebWorkSpaceActivity.this, 3));

        adapter = new WebWorkGroupAdapter(groups);
        adapter.setOnItemChildClickListener(new BaseQuickAdapter.OnItemChildClickListener() {
            @Override
            public void onItemChildClick(BaseQuickAdapter adapter, View view, int position) {
                GroupWebViewerActivity.start(mContext,groups.get(position).getInfos());
                startActivity(new Intent(mContext,PicInPicMarkDownActivity.class)
                        .addFlags(Intent.FLAG_ACTIVITY_LAUNCH_ADJACENT| Intent.FLAG_ACTIVITY_NEW_TASK)
                );
            }
        });
        mRecycler.setAdapter(adapter);

    }


    public static void debugStart(Context context) {
        ArrayList<WebInfo> infos = new ArrayList<>();
        infos.add(new WebInfo("https://www.baidu.com", "百度"));
        infos.add(new WebInfo("https://www.google.com", "谷歌"));
        infos.add(new WebInfo("https://www.baidu.com", "百度"));
        infos.add(new WebInfo("https://www.google.com", "谷歌"));
        infos.add(new WebInfo("https://www.baidu.com", "百度"));
        infos.add(new WebInfo("https://www.google.com", "谷歌"));
        WebWorkGroupAdapter.WebWorkGroupInfo group = new WebWorkGroupAdapter.WebWorkGroupInfo(
                "搜索器", infos
        );
        context.startActivity(new Intent(context, WebWorkSpaceActivity.class)
                .putExtra("groups", group)
        );
    }
}

File: app/src/main/java/edu/tjrac/swant/todo/view/WebWorkSpaceActivity.kt
package edu.tjrac.swant.todo.view

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.GridLayoutManager
import com.yckj.baselib.common.base.BaseActivity
import edu.tjrac.swant.todo.adapter.WebWorkGroupAdapter
import edu.tjrac.swant.todo.bean.WebInfo
import edu.tjrac.swant.unicorn.R
import kotlinx.android.synthetic.main.activity_web_work_space.*
import java.util.*

class WebWorkSpaceActivity : BaseActivity() {

    internal var groups: MutableList<WebWorkGroupAdapter.WebWorkGroupInfo> = ArrayList()
    internal var adapter: WebWorkGroupAdapter?=null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_web_work_space)
        val webs = intent.getParcelableExtra<WebWorkGroupAdapter.WebWorkGroupInfo>("groups")
        if (webs != null) {
            groups.add(webs)
        }
        recycler.setLayoutManager(GridLayoutManager(this@WebWorkSpaceActivity, 3))

        adapter = WebWorkGroupAdapter(groups)
        adapter!!.setOnItemChildClickListener { adapter, view, position ->
            GroupWebViewerActivity.start(mContext, groups[position].infos)
            startActivity(Intent(mContext, PicInPicMarkDownActivity::class.java)
                    .addFlags(Intent.FLAG_ACTIVITY_LAUNCH_ADJACENT or Intent.FLAG_ACTIVITY_NEW_TASK)
            )
        }
        recycler.setAdapter(adapter)

    }

    companion object {


        fun debugStart(context: Context) {
            val infos = ArrayList<WebInfo>()
            infos.add(WebInfo("https://www.baidu.com", "百度"))
            infos.add(WebInfo("https://www.google.com", "谷歌"))
            infos.add(WebInfo("https://www.baidu.com", "百度"))
            infos.add(WebInfo("https://www.google.com", "谷歌"))
            infos.add(WebInfo("https://www.baidu.com", "百度"))
            infos.add(WebInfo("https://www.google.com", "谷歌"))
            val group = WebWorkGroupAdapter.WebWorkGroupInfo(
                    "搜索器", infos
            )
            context.startActivity(Intent(context, WebWorkSpaceActivity::class.java)
                    .putExtra("groups", group)
            )
        }
    }
}
